File: gps/src/main/java/com/semid/gps/GpsSession.java
package com.semid.gps;

import android.content.Context;
import android.content.SharedPreferences;
import android.location.Location;

public class GpsSession {
	private String KEY_LAST_LAT = "last lat";
	private String KEY_LAST_LNG = "last lng";

	private SharedPreferences preferences;
	private SharedPreferences.Editor editor;

	public GpsSession(Context context) {
		preferences = context.getSharedPreferences("com.semid.gps", Context.MODE_PRIVATE);
	}

	public Location getLastLocation() {
		Location loc = new Location("passive");
		loc.setLatitude(Double.parseDouble(preferences.getString(KEY_LAST_LAT, "0")));
		loc.setLongitude(Double.parseDouble(preferences.getString(KEY_LAST_LNG, "0")));
		return loc;
	}

	public void setLastLocation(Location location) {
		editor = preferences.edit();
		editor.putString(KEY_LAST_LAT, String.valueOf(location.getLatitude()));
		editor.putString(KEY_LAST_LNG, String.valueOf(location.getLongitude()));
		editor.apply();
	}
}

File: gps/src/main/java/com/semid/gps/GpsSession.kt
package com.semid.gps

import android.content.Context
import android.content.SharedPreferences
import android.location.Location
import androidx.core.content.edit

class GpsSession private constructor() {
    private lateinit var preferences: SharedPreferences

    fun init(context: Context) {
        preferences = context.getSharedPreferences("com.semid.gps", Context.MODE_PRIVATE)
    }

    var lastLocation: Location
        get() {
            val loc = Location("passive")
            loc.latitude = preferences.getString(KEY_LAST_LAT, "0")!!.toDouble()
            loc.longitude = preferences.getString(KEY_LAST_LNG, "0")!!.toDouble()
            return loc
        }
        set(location) {
            preferences.edit {
                putString(KEY_LAST_LAT, location.latitude.toString())
                putString(KEY_LAST_LNG, location.longitude.toString())
            }
        }

    companion object {
        @Volatile
        private var INSTANCE: GpsSession? = null
        private const val KEY_LAST_LAT = "last lat"
        private const val KEY_LAST_LNG = "last lng"

        fun getInstance(context: Context): GpsSession {
            return INSTANCE ?: synchronized(this) {

                val instance = GpsSession()
                instance.init(context)

                INSTANCE = instance
                instance
            }
        }
    }
}